### PR TITLE
Merge masttro-15 to main

### PR DESCRIPTION
**Pull request by izztsfn**
This branch contains the script masttro_db_opportunity that adds two tables and also troubleshoots the cost field for cost per opportunity calculations

### CODE REVIEW

File: Clients/Masttro/Analytics Script/masttro_db_opportunity.sql [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    LEFT JOIN cost_aggregate campaign 
+    ON CONCAT(campaign_name._date,campaign_name._campaignName) = CONCAT(campaign._date,campaign._campaignName)  
+      
+    LEFT JOIN count_opps opps ON CONCAT(campaign_name._date,campaign_name._campaignName) = CONCAT(opps._date,opps._campaignName)  
+    --WHERE campaign_name._campaignName = '2309-Masttro-Branded-Keywords'
+  )
+
+  SELECT *, 
+  CASE
```
Comment (nazzatul21) [inline Clients/Masttro/Analytics Script/masttro_db_opportunity.sql:129]: Change to if statement , IF(_count_opp > 0 ,  SUM(_total_cost) OVER (PARTITION BY _date,_campaignName) / SUM(_count_opp) OVER (PARTITION BY _date,_campaignName), 0)
  AS _cost

File: Clients/Masttro/Analytics Script/masttro_db_opportunity.sql [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    _dealType,	
+    _companyName,	
+    _companyIndustry,	
+    _prospectID,
+    _name,	
+    _lifecycleStage	
+  )
+
+  WITH opp AS (
+    WITH contacts AS (
```
Comment (nazzatul21) [inline Clients/Masttro/Analytics Script/masttro_db_opportunity.sql]: make into single subquery, 

with contacts AS (
.
.
.
), ... AS (
.
.
.
), opportunity

**Review by nazzatul21 — CHANGES_REQUESTED** [review]

File: Clients/Masttro/Analytics Script/masttro_db_opportunity.sql (hunk 1 above)
Comment (izztsfn) [reply Clients/Masttro/Analytics Script/masttro_db_opportunity.sql]: Updated the CASE statement to IF

File: Clients/Masttro/Analytics Script/masttro_db_opportunity.sql (hunk 2 above)
Comment (izztsfn) [reply Clients/Masttro/Analytics Script/masttro_db_opportunity.sql]: This is the old commit before the formatting. It has already been changed in the latest commit where ive updated the formatting

**Comment by nazzatul21** [comment]
Approve. Good

**Review by nazzatul21 — APPROVED** [review]

**Pull request merged**